backend/src/util/VerifyFIN.ts

```diff
@@ -0,0 +1,34 @@
+const checkAlpha = ['X', 'W', 'U', 'T', 'R', 'Q', 'P', 'N', 'M', 'L', 'K'];
+const weightArray = [2, 7, 6, 5, 4, 3, 2];
+
+function validateFIN(fin: string): string {
+  const finUpper = fin.trim.toUpperCase();
+  if (finUpper.length === 9) {
+    const sumArray = fin.slice(1, 8).split('').map(Number); // FIN String into array of numbers to multiply by weight
+
+    let sum = 0;
+    for (let i = 0; i < sumArray.length; i += 1) {
+      sum += sumArray[i] * weightArray[i];
+    } // Cross multiply with weightArray to get sum
+
+    if (finUpper[0] === 'G') {
```

Comment:
I'm wary of this magic constant. What other values can the first letter be? Do they have any special behavior?

Reply:
The FIN structure is as such, @XXXXXXX#:
where FIN issued before 1 Jan 2000 are assigned the letter "F" and if it's after 1 Jan 2000, it is are assigned the letter "G". They do have a special behaviour that affects the last character, '#'.

For example, F00000001U and G0000001P will have the same sum when cross multiplying sumArray and weightArray. Therefore, to differentiate between the two, there is a shift of +4 to ensure the last character is different.

Reply:
I see. I'll need two things then:

- Cite a source for this
- Capture this information in a code comment so that nobody has to ask this again